File: internal/transport/http/responses/collection.go
package responses

import (
	"time"

	"github.com/google/uuid"
)

type Collection struct {
	ID        uuid.UUID `json:"id"`
	CreatedAt time.Time `json:"createdAt"`
	Name      string    `json:"name"`
}

type CollectionItem struct {
	ID           uuid.UUID `json:"id"`
	CreatedAt    time.Time `json:"createdAt"`
	CollectionID uuid.UUID `json:"collectionId"`
	Content      IContent  `json:"content"`
}

type BatchCollectionItems struct {
	*Paginated[*CollectionItem]
	CollectionID uuid.UUID `json:"collectionId"`
}

File: apps/web/lib/model/collection.ts
import { Content } from './content'
import { Paginated } from './types'

export type Collection = {
  id: string
  createdAt: Date
  name: string
}

export type CollectionItem = {
  id: string
  createdAt: Date
  collectionId: string
  content: Content
}

export type BatchCollectionItems = Paginated<CollectionItem> & {
  collectionId: string
}
